code/modules/atmospherics/portable/portable_pressurizer.dm

```diff
@@ -0,0 +1,429 @@
+					if("viscerite")
+						GM.nitrogen += 1500 * progress / 100
+					if("char")
+						GM.carbon_dioxide += 500 * progress / 100
+					if("plasmastone")
+						GM.toxins += 500 * progress / 100
+			else
+				playsound(src.loc, 'sound/machines/buzz-two.ogg', 20)
+				process_materials = PROCESS_PAUSED
+				target_material?.set_loc(src.loc)
+				target_material = null
+				return
+			src.air_contents.merge(GM)
+			material_progress = clamp(material_progress + progress, 0, 100)
+			use_power(src.powconsumption)
+		else
+			qdel(target_material)
+			target_material = null
+
+	emag_act(var/mob/user, var/obj/item/card/emag/E)
+		if (!src.emagged)
+			if (user)
+				user.show_text("You short out the material processor on [src].", "red")
+			src.audible_message("<span class='combat'><B>[src] buzzes oddly!</B></span>")
+			playsound(src.loc, "sparks", 50, 1, -1)
+			whitelist += blacklist
+			src.emagged = TRUE
+			return 1
+		return 0
+
+	demag(var/mob/user)
+		if (!src.emagged)
+			return 0
+		if (user)
+			user.show_text("You repair [src]'s material processor.", "blue")
+		src.emagged = FALSE
+		src.process_rate = min(src.process_rate, 3)
+		whitelist = initial(whitelist)
+		return 1
+
+	// attack by item places it in to disposal
+	attackby(obj/item/I, mob/user)
+		if(status & BROKEN)
+			return
+		if(iswrenchingtool(I) || istype(I,/obj/item/device/analyzer/atmospheric) || istype(I,/obj/item/card/emag))
+			..()
+			return
+		if(istype(I,/obj/item/electronics/scanner) || istype(I,/obj/item/deconstructor) || (istype(I,/obj/item/device/pda2)))
+			user.visible_message("<span class='alert'><B>[user] hits [src] with [I]!</B></span>")
+			return
+		if (istype(I,/obj/item/satchel/) && I.contents.len)
+			var/action = input(user, "What do you want to do with the satchel?") in list("Place it in the Chute","Empty it into the Chute","Never Mind")
+			if (!action || action == "Never Mind")
+				return
+			if (!in_interact_range(src, user))
+				boutput(user, "<span class='alert'>You need to be closer to the chute to do that.</span>")
+				return
+			if(action == "Empty it into the Chute")
+				var/obj/item/satchel/S = I
+				for(var/obj/item/O in S.contents) O.set_loc(src)
+				S.satchel_updateicon()
+				user.visible_message("<b>[user.name]</b> dumps out [S] into [src].")
+				return
+		if (istype(I,/obj/item/storage/) && I.contents.len)
+			var/action = input(user, "What do you want to do with [I]?") as null|anything in list("Place it in the Chute","Empty it into the chute","Never Mind")
+			if (!action || action == "Never Mind")
+				return
+			if(!in_interact_range(src, user))
+				boutput(user, "<span class='alert'>You need to be closer to the chute to do that.</span>")
+				return
+			if(action == "Empty it into the chute")
+				var/obj/item/storage/S = I
+				for(var/obj/item/O in S)
+					O.set_loc(src)
+					S.hud.remove_object(O)
+				user.visible_message("<b>[user.name]</b> dumps out [S] into [src].")
+				return
+			if(isnull(action)) return
```

Comment:
Would remove the "place it into the chute" copy-pasta.

Reply:
i r dum dum